app/src/main/java/org/simple/clinic/teleconsultlog/prescription/TeleconsultPrescriptionEffectHandler.kt

```diff
@@ -27,9 +37,30 @@ class TeleconsultPrescriptionEffectHandler @AssistedInject constructor(
         .addAction(GoBack::class.java, uiActions::goBackToPreviousScreen, schedulersProvider.ui())
         .addAction(ShowSignatureRequiredError::class.java, uiActions::showSignatureRequiredError, schedulersProvider.ui())
         .addTransformer(LoadDataForNextClick::class.java, loadDataForNextClick())
+        .addTransformer(CreatePrescription::class.java, createPrescription())
         .build()
   }
 
+  private fun createPrescription(): ObservableTransformer<CreatePrescription, TeleconsultPrescriptionEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .observeOn(schedulersProvider.io())
+          .doOnNext { medicalRegistrationIdPreference.set(Optional.of(it.medicalRegistrationId)) }
+          .doOnNext { teleconsultRecordRepository.updateMedicalRegistrationId(it.teleconsultRecordId, it.medicalRegistrationId) }
+          .doOnNext { updatePrescribedDrugTeleconsultId(it.patientUuid, it.teleconsultRecordId) }
+          .map { PrescriptionCreated(it.medicalInstructions) }
+    }
+  }
+
+  private fun updatePrescribedDrugTeleconsultId(patientUuid: UUID, teleconsultRecordId: UUID) {
+    val prescribedDrugs = prescriptionRepository.newestPrescriptionsForPatientImmediate(patientUuid)
+
+    prescriptionRepository.updateTeleconsultationIds(
+        drugUuids = prescribedDrugs.map { it.uuid },
```

Comment:
We could move this `map {}` to the place where we are reading the prescriptions from the repository since we don't use the prescribed drugs anywhere else.

Reply:
Yep, definitely. I was just afraid if it was going to be some hidden behavior